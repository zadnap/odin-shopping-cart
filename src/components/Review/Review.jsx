import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import styles from './Review.module.scss';
import {
  faChevronLeft,
  faChevronRight,
  faStar,
} from '@fortawesome/free-solid-svg-icons';
import Button from '@/components/Button/Button';
import ErrorMessage from '@/components/ErrorMessage/ErrorMessage';

function Review({ numberOfReview, currentReview, onNext, onPrev }) {
  return (
    <section className={styles.review}>
      <div className={styles.header}>
        <h1 className={styles.title}>Reviews ({numberOfReview})</h1>
        {numberOfReview > 0 && (
          <div className={styles.navigator}>
            <Button
              square
              aria-label="View Previous Review Article"
              onClick={onPrev}
            >
              <FontAwesomeIcon icon={faChevronLeft} />
            </Button>
            <Button
              square
              aria-label="View Next Review Article"
              onClick={onNext}
            >
              <FontAwesomeIcon icon={faChevronRight} />
            </Button>
          </div>
        )}
      </div>
      {numberOfReview > 0 ? (
        <article className={styles.reviewArticle}>
          <div className={styles.info}>
            <img
              className={styles.profileImage}
              src={currentReview.profileImage}
              alt={`${currentReview.author}'s profile image`}
            />
            <div className={styles.authorInfo}>
              <h2 className={styles.name}>
                A review by {currentReview.author}
              </h2>
              <p className={styles.releaseDate}>{currentReview.releaseDate}</p>
            </div>
            <div className={styles.rating}>
              <FontAwesomeIcon icon={faStar} />{' '}
              <span>{currentReview.rating}%</span>
            </div>
          </div>
          <p
            className={styles.detail}
            dangerouslySetInnerHTML={{ __html: currentReview.detail }}
          />
        </article>
      ) : (
        <ErrorMessage message="This movie has not received any reviews yet" />
      )}
    </section>
  );
}

export default Review;
