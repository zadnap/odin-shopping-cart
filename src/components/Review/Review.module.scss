@use '@/styles/variables.scss' as *;

.review {
  .header {
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
    margin-bottom: 10px;

    .title {
      font-size: 1.8rem;
    }
    .navigator {
      display: flex;
      gap: 6px;
    }
  }
  .reviewArticle {
    padding: 15px;
    background-color: $subbackground-color;
    border-radius: $border-radius;
    margin-bottom: 10px;

    .info {
      display: flex;
      align-items: center;
      margin-bottom: 15px;
      gap: var(--gap);

      .profileImage {
        width: 50px;
        height: 50px;
        border-radius: $border-radius;
        object-fit: cover;
      }
      .authorInfo {
        display: flex;
        flex-direction: column;
        .name {
          font-size: 1.6rem;
          font-weight: 600;
        }
        .releaseDate {
          font-size: 1.3rem;
        }
      }
      .rating {
        font-size: 1.4rem;
        padding: 4px 7px;
        align-self: self-start;
        border-radius: $border-radius;
        background-color: $overlay-color;
        color: yellow;
        margin-left: auto;
      }
    }

    .detail {
      white-space: pre-line;

      a {
        color: $text-color;
        font-weight: 500;
        text-decoration: underline;
      }
    }
  }
}

@media (max-width: $mobile-max) {
  .review .reviewArticle .info {
    flex-direction: column;
    gap: var(--gap);

    .authorInfo {
      align-items: center;
    }

    .rating {
      margin: 0 auto;
    }
  }
}
