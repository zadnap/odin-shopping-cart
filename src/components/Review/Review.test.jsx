import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import Review from './Review';

const mockProps = {
  profileImageSrc:
    'https://secure.gravatar.com/avatar/bf3b87ecb40599290d764e6d73c86319.jpg?s=45',
  name: 'Chris Sawin',
  releaseDate: 'June 1, 2025',
  rating: 60,
  detail: 'While watching Disney',
};

describe('Review component', () => {
  it('should render to match snapshot', () => {
    const { container } = render(<Review {...mockProps} />);

    expect(container).toMatchSnapshot();
  });
});
