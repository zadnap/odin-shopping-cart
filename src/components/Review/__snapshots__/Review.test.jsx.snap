// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Review component > should render to match snapshot 1`] = `
<div>
  <section
    class="_review_031c44"
  >
    <div
      class="_header_031c44"
    >
      <h1
        class="_title_031c44"
      >
        Reviews (
        )
      </h1>
    </div>
    <div
      class="_errorMessage_2e14f5"
      role="alert"
    >
      <img
        alt=""
        class="_errorImage_2e14f5"
        src="/src/assets/error-face.png"
      />
      <p
        class="_message_2e14f5"
      >
        This movie has not received any reviews yet
      </p>
    </div>
  </section>
</div>
`;
